src/Stack/Types/Build.hs

```diff
@@ -448,6 +449,7 @@ defaultBuildOpts = BuildOpts
     , boptsPreFetch = False
     , boptsBuildSubset = BSAll
     , boptsFileWatch = False
+    , boptsFileWatchPoll = False
```

Comment:
It's a bit unclear what the semantics are when both `boptsFileWatch` and `boptsFileWatchPoll` are set. I'd probably restructure this a bit as:

```haskell
data FileWatch
  = NoFileWatch
  | FileWatch
  | FileWatchPoll

boptsFileWatch :: FileWatch
```

Then when parsing command line arguments, you can use the `Alternative` instance and get something like:

```haskell
flag' FileWatch (long "file-watch" <> ...) <|> flag' FileWatchPoll (long "file-watch-poll" <> ...)
  <|> pure NoFileWatch
```